app/controllers/api/v1/base_controller.rb

```diff
@@ -0,0 +1,19 @@
+module Api
+  module V1
+    class BaseController < ApplicationController
+
+      before_filter :restrict_access
+      
+      private
+      
+      def restrict_access
+        unless ApiKey.exists?(access_token: params[:token])
+          authenticate_or_request_with_http_token do |token, options|

```

Comment:
Unused block argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used.